packages/anoncreds-rs/src/services/AnonCredsRsHolderService.ts

```diff
@@ -48,31 +51,35 @@ export class AnonCredsRsHolderService implements AnonCredsHolderService {
     agentContext: AgentContext,
     options?: CreateLinkSecretOptions
   ): Promise<CreateLinkSecretReturn> {
+    let masterSecret: MasterSecret | undefined
     try {
+      masterSecret = MasterSecret.create()
+
+      // FIXME: This is a very specific format of anoncreds-rs. I think it should be simply a string
+      const linkSecretJson = masterSecret.toJson() as { value: { ms: string } }
+
       return {
         linkSecretId: options?.linkSecretId ?? utils.uuid(),
-        linkSecretValue: JSON.parse(MasterSecret.create().toJson()).value.ms,
+        linkSecretValue: linkSecretJson.value.ms,
       }
-    } catch (error) {
-      agentContext.config.logger.error(`Error creating Link Secret`, {
-        error,
-      })
-      throw new AnonCredsRsError('Error creating Link Secret', { cause: error })
+    } finally {
+      masterSecret?.handle.clear()
     }
   }
 
   public async createProof(agentContext: AgentContext, options: CreateProofOptions): Promise<AnonCredsProof> {
     const { credentialDefinitions, proofRequest, selectedCredentials, schemas } = options
 
+    let presentation: Presentation | undefined
```

Comment:
```suggestion
    let presentation: Presentation
```

Should be fine I believe.

Reply:
I had to add it because I'm clearing it in the finally block (without it, it gives a `Variable 'presentation' is used before being assigned`)

Reply:
Ahh yeah makes sense.